### PR TITLE
bootstraph.sh: Fail early if go get fails.

### PR DESCRIPTION
**Pull request by michael-berlin**
Recently, we saw some Travis failures because not all Go dependencies
were successfully 'go get'd. However, the test run didn't stop early and
instead it was difficult to find out the original reason for the test
failure. This commit will make it easier to find out when the bootstrap
flaked and the test simply has to be restarted.

@enisoc @alainjobart

### CODE REVIEW

File: bootstrap.sh [hunk 1]
```diff
@@ -9,21 +9,30 @@ if [ "$1" = "--skip_root_installs" ]; then
   SKIP_ROOT_INSTALLS=True
 fi
 
+# Exits if $1 is not 0 and prints $2 as error message.
```
Comment (enisoc) [inline bootstrap.sh]: We had a pattern before where there's a "fail" function that prints the message and exits. You would use it like this:

``` sh
do something || fail "exit message"
```

I think this is more idiomatic BASH (if there is such a thing).

File: bootstrap.sh [hunk 2]
```diff
@@ -9,21 +9,30 @@ if [ "$1" = "--skip_root_installs" ]; then
...
+    rc=1
+    msg="missing arguments in fail_on_error() call in bootstrap.sh."
+  fi
+  if [ $rc -ne 0 ]; then
+    echo "ERROR: $msg"
+    exit $rc
+  fi
+}
+
 if [ ! -f bootstrap.sh ]; then
```
Comment (enisoc) [inline bootstrap.sh]: As mentioned above, this can become:

``` sh
[ -f bootstrap.sh ] || fail "blah blah"
```

Comment (michael-berlin) [inline bootstrap.sh]: Why do we need double brackets? Single brackets seem to be fine.

File: bootstrap.sh (hunk 1 above)
Comment (michael-berlin) [inline bootstrap.sh]: I've changed it as you proposed.

PTAL

File: bootstrap.sh [hunk 3]
```diff
@@ -47,10 +42,7 @@ else
     (cd zookeeper-$zk_ver/src/c && \
     ./configure --prefix=$zk_dist && \
     make install) && rm -rf zookeeper-$zk_ver zookeeper-$zk_ver.tar.gz)
-  if [ $? -ne 0 ]; then
-    echo "zookeeper build failed"
-    exit 1
-  fi
+  [ $? -eq 0 ] || "zookeeper build failed"
```
Comment (enisoc) [inline bootstrap.sh]: Forgot the "fail". Also you could just do `...tar.gz) || fail "zookeeper build failed"`, but maybe that's more confusing.

File: bootstrap.sh (hunk 2 above)
Comment (enisoc) [inline bootstrap.sh]: You're right. Not sure what I did wrong before when testing it with single brace, but it works.

File: bootstrap.sh (hunk 3 above)
Comment (michael-berlin) [inline bootstrap.sh]: Fixed.

I prefer to keep it on a separate line for readability :-)

**Comment by enisoc** [comment]
LGTM

**Pull request merged**